### PR TITLE
Trying pointers and errors

### PR DESCRIPTION
**Pull request by rjar2020**
- Using pointers
- Handling errors, the basics.
- Including a linter to enforce error handling, `errcheck`

### CODE REVIEW

**Comment by rjar2020** [comment]
Linter error fixed, everything green.

File: pointer/wallet_test.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+			fmt.Printf("%s\" ", err)
+		}
+	}
+}
+
+func BenchmarkWithdrawInsufficientFunds(b *testing.B) {
+	wallet := Wallet{balance: Bitcoin(10)}
+	for i := 0; i < b.N; i++ {
+		err := wallet.Withdraw(Bitcoin(100))
+		if err != nil {
```
Comment (rjar2020) [inline pointer/wallet_test.go:81]: My homage to an empty `if` @JuanCalcagno416

File: pointer/wallet.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	w.balance += amount
+}
+
+//Balance allows a client to query a wallet's balance
+func (w *Wallet) Balance() Bitcoin {
+	return w.balance
+}
+
+//ErrInsufficientFunds is thronw when a withdraw can't be done in a wallet as there aren't enough funds
+var ErrInsufficientFunds = errors.New("cannot withdraw, insufficient funds")
```
Comment (JuanCalcagno416) [inline pointer/wallet.go]: Maybe we can put the error below the imports? :eyes:

**Review by JuanCalcagno416 — APPROVED** [review]
LGTM. Left a dumb comment and also you can squash the commits before merging. BTW I like empty `if`  :trollface:

File: pointer/wallet.go (hunk 2 above)
Comment (rjar2020) [reply pointer/wallet.go]: Thanks Nacho! I didn't want to go full Java pushing this to the top, lets fix it!

**Pull request merged**